src/entities.ts

```diff
@@ -115,10 +126,34 @@ customEvents.on('gameLoaded', () => {
     }
   }
 
-  bot.on('entitySpawn', entityData)
+  bot.on('entitySpawn', (e) => {
+    entityData(e)
+    if (appViewer.playerState.cameraEntity === e.id) {
+      updateCamera(e)
+    }
+  })
   bot.on('entityUpdate', entityData)
   bot.on('entityEquip', entityData)
 
+  bot._client.on('camera', (packet) => {
+    if (bot.player.entity.id === packet.cameraId) {
+      if (appViewer.playerState.cameraEntity) {
+        const entity = bot.entities[appViewer.playerState.cameraEntity]
+        appViewer.playerState.cameraEntity = undefined
+        if (entity) {
+          appViewer.backend?.updateEntity(entity)
```

Comment:
why not entityData function call here? This method seems redundant

Reply:
As far as I can tell `entityData` seems to do something different. (and seems to only really do something for players?) The updating here is necessary so that the entity is hidden when it is the one the player spectates.

Reply:
This function will emit the entity update then https://github.com/zardoy/minecraft-web-client/blob/1599917134a940d8f7ce2e386115ab5193bc9ebc/renderer/viewer/lib/worldDataEmitter.ts#L124